### DISCUSSION TITLE
Fan Control On/Off Toggle Sensor

### DISCUSSION THREAD

**Discussion by NitroMehmet** [Rem0o/FanControl.Releases#1960] [Ideas]
It would be better if a switch on/off fan control sensor added. What I mean normally fan controls handled by hardware itself but sometimes I want to Fan Control kick in and override hardware controlled fan curve.

In my situation by default GPU fans are controlled by GPU itself. I don't mind to apply custom fan curve for them in FanControl. They are silent and well optimized by factory default. However my Gen.4 M2 SSD is under my GPU. When I do daily desktop usage my GPU is idle and fans are not spinning most of the time (below 60 C). This builds up a warm chamber around my GPU and unfortunately my SSD overheats when I do rapid squential read/writes.

What I want can developer add a new mix sensor that toggles on/off switch for certain fans?

![Screenshot 2023-08-03 0350467](https://github.com/Rem0o/FanControl.Releases/assets/56390583/f6f611b2-9b8e-4ab5-9ad6-481bcf41eeab)

**Comment by Rem0o** [comment]
Why don't you simply make a curve for your GPU from a mix that includes your m.2? Just replicate the stock curve the best you can so you'll get the same behavior.  No need to add extra complexity.

**Comment by NitroMehmet** [comment]
I tried but it wasn't silent as card manages fans itself as. On top of that I think the GPU handles fan speeds depending on hotspot temperature. I play games but also I render and encode with my GPU that stresses memory. When core temperature never exceed 60C threshold but the memory does. This may shorten the GPU lifespan as if memory runs hot but fans not spin. Sometimes when GPU core temp exceed threshold but fans not start to spin. It seems GPU also measures power consumption to detect load. I inspect fan speed behaviour in long term and it's complicated. I couldn't simulate nor managed to export fan profile for GPU. 
TLDR; I don't want to shorten lifespan fan bearings nor don't want to early failure due to overheat.

I agree with you no need to add extra complexity. Also using GPU fans to create air circulation to cooldown SSD is not a solution but more like a turn around. Bet it would be fine adding toggle sensor as a plugin if it's possible🤷‍♂️
![Screenshot 2023-08-05 001427](https://github.com/Rem0o/FanControl.Releases/assets/56390583/416d99bb-aa79-4933-9569-49500aa22208)

**Reply by Rem0o** [reply]
One thing you could do is use the nvidia override checkbox and set a curve at 0 until you want to control it from fan control.   That 0 area is interpreted then as "turn the control off".

**Reply by NitroMehmet** [reply]
that worked, thanks 👍👍
